### PR TITLE
python formulae: pip, setuptools, pkgconfig linkage.

### LINKED ISSUE #38121: python3 and pkg-config
I've installed python3using brew, but it seems that `pkg-config` isn't picking up the .pc files that are placed in `/Library/Frameworks/Python.framework/Versions/3.4/lib/pkgconfig`.

I'm not a brew master, nor do I know about the details of python, but I would have imagined that something like the following line (from https://github.com/Homebrew/homebrew-x11/blob/master/hexchat.rb) should be present in the brew file:

    ENV.append_path "PKG_CONFIG_PATH", "#{HOMEBREW_PREFIX}/Frameworks/Python.framework/Versions/#{py_ver}/lib/pkgconfig/

I can't tell if that's me suggesting something wrong, or if it's just missing from https://github.com/Homebrew/homebrew/blob/master/Library/Formula/python3.rb

Does anybody know?

### PR DESCRIPTION
**Pull request by DomT4**
Bumped Pip, Bumped Setuptools, Linked the pkgconfig files so they're available automatically.

Closes #38121

### CODE REVIEW

**Comment by DomT4** [comment]
CC @tdsmith

File: Library/Formula/python3.rb [hunk 1]
```diff
@@ -142,6 +142,9 @@ def install
     ["Headers", "Python", "Resources"].each { |f| rm(prefix/"Frameworks/Python.framework/#{f}") }
     rm prefix/"Frameworks/Python.framework/Versions/Current"
 
+    # Symlink the pkgconfig files into HOMEBREW_PREFIX so they're accessible.
+    (lib+"pkgconfig").install_symlink Dir["#{frameworks}/Python.framework/Versions/3.4/lib/pkgconfig/*"]
```
Comment (tdsmith) [inline Library/Formula/python3.rb]: Use `/` instead of `+` for local consistency and #{xy} instead of 3.4

**Comment by tdsmith** [comment]
lgtm modulo nits, thanks for grabbing this

File: Library/Formula/python3.rb (hunk 1 above)
Comment (DomT4) [inline Library/Formula/python3.rb]: Is Python.rb okay using ` Current ` still or should that be ` #{xy} ` as well?

Comment (DomT4) [inline Library/Formula/python3.rb]: Fixed up ` + ` into `/`. Changed Python3's ` 3.4 ` into `#{xy}`. Didn't change ` Current ` on `python.rb`. Just poke if that needs changing to anything else as well.

**Comment by DomT4** [comment]
No worries! Happy to help on the bits I can. Thanks for the speedy feedback, appreciated!

File: Library/Formula/python3.rb (hunk 1 above)
Comment (tdsmith) [inline Library/Formula/python3.rb]: yeah, Current is fine for 2.7, thanks!

**Comment by DomT4** [comment]
Weird. ` sip `'s test failed. Will look into that early this afternoon.

**Comment by tdsmith** [comment]
Nah, this is good; it's a test-bot thing, because sip is tested against python3 after python is uninstalled.

**Comment by DomT4** [comment]
Ah! Thanks! You probably just saved me an hour of looking quite confused this afternoon :smile_cat:.

**Pull request closed without merging**